lib/backend/k8s/resources/watcher.go

```diff
@@ -36,28 +36,31 @@ func newK8sWatcherConverter(
 	name string,
 	converter ConvertK8sResourceToKVPair,
 	k8sWatch kwatch.Interface,
+	versionconverter VersionConverter,
```

Comment:
You shouldn't need to change this code I think.

Even without my suggestion above ^^^ , this helper module already includes a converter and the version conversion could have just been included in that by the consumer of this watcher helper.